File: tarea1/verArreglo.java
import java.util.Arrays;
public class verArreglo {
    public static void main(String[] args) {
        int[] arreglo = {20,30,40,50,60,70};
        if(args.length == 0) {
            System.out.println(Arrays.toString(arreglo));
            return;
        }
        switch (args[0]) {
            case "recorrido": 
            {
                int suma = 0;
                for(int i = 0; i < arreglo.length; i++) {
                    System.out.println("En la posicion " + (i + 1) + " del arreglo se encuentra el valor: " + arreglo[i] + ".");
                    suma += arreglo[i];
                }
                System.out.println("La sumatoria es: " + suma + ".");
            break;
            }
            case "reversa": {
                int suma = 0;
                for(int i = arreglo.length - 1; i >= 0; i--) {
                    System.out.println("En la posicion " + (i + 1) + " del arreglo se encuentra el valor: " + arreglo[i] + ".");
                    suma += arreglo[i];
                }
                System.out.println("La sumatoria es: " + suma + ".");
            break;
            }
            case ("indice"):
                if(args.length == 1){
                    System.out.println("Por favor introduzca un indice.");
                } else if (Integer.parseInt(args[1]) > 0 && Integer.parseInt(args[1]) <= arreglo.length) {
                    System.out.println("En la posicion " + args[1] + " del arreglo se encuentra el valor: " + arreglo[Integer.parseInt(args[1]) - 1] + ".");
                } else {
                    System.out.println("Indice invalido.");
                    System.out.println("Prueba con un valor del 1 al " + arreglo.length + ".");
                }
            break;
            case "buscar":
                if (args.length == 1) {
                    System.out.println("Por favor introduzca un valor a buscar.");
                } else {
                int indice = -1;
                for (int i = 0; i < arreglo.length; i++) {
                    if (arreglo[i] == Integer.parseInt(args[1])){
                        indice = i + 1;
                    }
                }
                if (indice == -1 && args.length == 2){
                        System.out.println("El valor no se encontro en el arreglo.");
                    } else {
                        System.out.println("El valor se encontro en el indice " + indice + ".");
                    }
                }
            break;
        }
    }
}

File: tarea1/cs/verArreglo.cs
using System;

class verArreglo
{
    static void Main(string[] args)
    {
        int[] arreglo = { 20, 30, 40, 50, 60, 70 };
        if (args.Length == 0)
        {
            Console.Write("[");
            for (int i = 0; i < arreglo.Length; i++)
            {
                Console.Write(arreglo[i]);
                if (i < arreglo.Length - 1)
                    Console.Write(", ");
            }
            Console.WriteLine("]");
        }
        else
        {
            switch (args[0])
            {
                case "recorrido":
                    {
                        int suma = 0;
                        for (int i = 0; i < arreglo.Length; i++)
                        {
                            Console.WriteLine($"En la posicion {i + 1} del arreglo se encuentra el valor: {arreglo[i]}.");
                            suma += arreglo[i];
                        }
                        Console.WriteLine($"La sumatoria es: {suma}.");
                        break;
                    }
                case "reversa":
                    {
                        int suma = 0;
                        for (int i = arreglo.Length - 1; i >= 0; i--)

                        {
                            Console.WriteLine($"En la posicion {i + 1} del arreglo se encuentra el valor: {arreglo[i]}.");
                            suma += arreglo[i];
                        }
                        Console.WriteLine($"La sumatoria es: {suma}.");
                        break;
                    }
                case "indice":
                    if (args.Length == 1)
                    {
                        Console.WriteLine("Por favor introduzca un indice.");
                    }
                    else if (int.Parse(args[1]) < 1 || int.Parse(args[1]) > arreglo.Length)
                    {
                        Console.WriteLine("Indice invalido.");
                        Console.WriteLine($"Prueba con un valor del 1 al {arreglo.Length}.");
                    }
                    else
                    {
                        Console.WriteLine($"En la posicion {args[1]} del arreglo se encuentra el valor {arreglo[int.Parse(args[1]) - 1]}.");
                    }
                    break;
                case "buscar":
                    if (args.Length == 1)
                    {
                        Console.WriteLine("Por favor introduzca un elemento a buscar.");
                    }
                    else
                    {
                        for (int i = 0; i < arreglo.Length; i++)
                        {
                            if (arreglo[i] == int.Parse(args[1]))
                            {
                                Console.WriteLine($"El valor se encontro en el indice {i + 1}.");
                                break;
                            } else if (i == arreglo.Length - 1)
                            {
                                Console.WriteLine("El valor no se encontro en el arreglo.");
                            }
                        }
                    }
                    break;
            }
        }
    }
}